src/smartdc/bin/sdc-factoryreset.sh

```diff
@@ -37,18 +38,22 @@ if [[ -n $1 ]] && [[ $1 = "--help" ]]; then
 	usage
 fi
 
-while getopts "h" opt
+final_command="reboot"
+final_verb="Rebooting"
+
+while getopts "hs" opt
 do
 	case "$opt" in
 		h)	usage;;
+		s)	final_command="poweroff"; final_verb="Powering off";;
 		*)	usage;;
 	esac
 done
 
 trap abort SIGINT
 
-printf "WARNING: This machine will reboot and destroy its ZFS pools after "
-printf "rebooting.\n"
+printf "WARNING: This machine will $final_command and destroy its ZFS pools "
+printf "after the next reboot.\n"
```

Comment:
Here, I would just say "after the next boot", which covers both boot from power off and reboot.